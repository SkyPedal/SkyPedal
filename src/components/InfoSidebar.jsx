import React from "react";

const InfoSidebar = () => {
  return (
    <div className="w-1/5 h-full sticky right-0 border-l-4">
      <p>Info Sidebar</p>
    </div>
  );
};

export default InfoSidebar;
