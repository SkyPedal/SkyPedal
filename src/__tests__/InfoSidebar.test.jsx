import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import InfoSidebar from "../components/InfoSidebar";
import { BrowserRouter as Router } from "react-router-dom";

describe("InfoSidebar", () => {
  it("snapshot renders", () => {
    const { asFragment } = render(
      <Router>
        <InfoSidebar />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
