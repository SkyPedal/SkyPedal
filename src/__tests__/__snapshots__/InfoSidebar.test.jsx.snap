// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoSidebar > snapshot renders 1`] = `
<DocumentFragment>
  <div
    class="hidden md:flex md:flex-col justify-evenly basis-52 h-full sticky right-0 border-l-4 border-solid"
    style="border-image: var(--sky-gradient-shape) 1;"
  >
    <div
      class="m-1 flex flex-col"
    >
      <h1
        class="text-center"
      >
        Impact
      </h1>
      <hr />
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        CO2 Saved
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        10kg
      </p>
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        Distance Travelled
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        10miles
      </p>
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        Hours Commuted
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        4 hours
      </p>
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        Calories Burned
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        500cal
      </p>
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        Money Saved
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        $10
      </p>
    </div>
    <div
      class="m-1 flex flex-col"
    >
      <h1>
        Leaderboard
      </h1>
      <hr />
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        Position (Week)
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        2nd
      </p>
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        Position (Month)
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        3rd
      </p>
    </div>
    <div
      class="m-1 flex flex-col"
    >
      <h1
        class="text-center"
      >
        Rewards
      </h1>
      <hr />
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        Points
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        100
      </p>
      <i
        class="text-sm text-left p-1 pl-4 text-gray-500"
      >
        Active Rewards
      </i>
      <p
        class="text-lg p-1 pb-0"
      >
        Reward 1
      </p>
      <p
        class="text-lg p-1 pb-0"
      >
        Reward 2
      </p>
      <p
        class="text-lg p-1 pb-0"
      >
        Reward 3
      </p>
    </div>
  </div>
</DocumentFragment>
`;
